src/uri/authority.rs

```diff
@@ -192,6 +192,64 @@ impl PartialEq<str> for Authority {
     }
 }
 
+impl PartialEq<Authority> for str {
+    fn eq(&self, other: &Authority) -> bool {
+        self.eq_ignore_ascii_case(other.as_str())
+    }
+}
+
+impl PartialEq<String> for Authority {
+    fn eq(&self, other: &String) -> bool {
+        self.data.eq_ignore_ascii_case(other.as_str())
+    }
+}
+
+impl PartialEq<Authority> for String {
+    fn eq(&self, other: &Authority) -> bool {
+        self.as_str().eq_ignore_ascii_case(other.as_str())
+    }
+}
+
+/// Case-insensitive ordering
+///
+/// # Examples
+///
+/// ```
+/// # use http::uri::Authority;
+/// let authority: Authority = "DEF.com".parse().unwrap();
+/// assert!(authority < *"ghi.com");
+/// assert!(authority > *"abc.com");
+/// ```
+impl PartialOrd for Authority {
+    fn partial_cmp(&self, other: &Authority) -> Option<cmp::Ordering> {
+        self.data.to_ascii_lowercase().partial_cmp(&other.data.to_ascii_lowercase())
```

Comment:
All of these comparisons actually allocate and copy a new string before comparing. It's probably faster to just compare the bytes individually. Something like:

```rust
let left = self.data.iter().map(|b| b.to_ascii_lowercase());
let right = other.data.iter().map(|b| b.to_ascii_lowercase());
Some(left.cmp(right))
```

Reply:
Nice, I'll give it a try.

Reply:
@seanmonstar I'm having issue with calling `.iter()` on the data (which I'll try to figure out).

I'm looking at the implementation of `to_ascii_lowercase` and it creates a vec and returns back (unsafe) a string. Since this proposal would still map all the bytes to another vec, wouldn't the cost be comparable?

Reply:
Oh, my bad, try `data.as_bytes().iter()`.

The difference is that the `to_ascii_lowercase` will reserve space in the heap, and then copy the bytes into that, before doing the comparisons, so doing it this way actually can be rather noticeable.

Reply:
oh, cool, that worked!